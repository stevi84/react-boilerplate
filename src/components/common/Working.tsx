import { CircularProgress, Stack } from '@mui/material';
import { useTranslation } from 'react-i18next';

interface WorkingProps {
  isReading: boolean;
  isSubmitting: boolean;
}

export const Working = (props: WorkingProps) => {
  const { isReading, isSubmitting } = props;

  const { t } = useTranslation();

  return (
    <Stack direction={'column'} justifyContent={'center'} alignItems={'center'} style={{ height: '100%' }}>
      <CircularProgress id="loading-spinner" />
      {isSubmitting ? t('submitting') : isReading ? t('reading') : null}
    </Stack>
  );
};
