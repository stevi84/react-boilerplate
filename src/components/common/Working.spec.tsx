import { describe, it, expect, vi } from 'vitest';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { render, screen } from '@testing-library/react';
import { Working } from './Working';

vi.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str: string) => str,
    i18n: {
      language: 'de',
      changeLanguage: vi.fn(),
    },
  }),
}));

describe('Working', () => {
  it('should equal saved snapshot', () => {
    const renderer: ShallowRenderer = createRenderer();
    renderer.render(<Working isReading={false} isSubmitting={true} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('should show reading', () => {
    render(<Working isReading={true} isSubmitting={false} />);
    expect(screen.getByText('reading')).toBeTruthy();
  });

  it('should show submitting', () => {
    render(<Working isReading={false} isSubmitting={true} />);
    expect(screen.getByText('submitting')).toBeTruthy();
  });

  it('should show submitting too', () => {
    render(<Working isReading={true} isSubmitting={true} />);
    expect(screen.getByText('submitting')).toBeTruthy();
  });
});
