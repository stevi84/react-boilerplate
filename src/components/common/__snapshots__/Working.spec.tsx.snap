// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Working > should equal saved snapshot 1`] = `
<ForwardRef(Grid)
  alignItems="center"
  direction="column"
  justifyContent="center"
  style={
    {
      "height": "100%",
    }
  }
>
  <ForwardRef(CircularProgress)
    id="loading-spinner"
  />
  submitting
</ForwardRef(Grid)>
`;
